### PR TITLE
California Investment interest expense deduction

### LINKED ISSUE #3837: Populate the California Investment interest expense deduction
[Source](https://www.ftb.ca.gov/forms/2021/2021-3526.pdf) 

The ca_investment_interest_deduction file will be added in #3275

### PR DESCRIPTION
**Pull request by vrathi101**
Fixes #3837

### CODE REVIEW

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        investment_expenses = person("investment_expense", period)  # Line 5
+        net_investment_income = (
+            investment_interest_expense - investment_expenses
+        )  # Line 6
+        investment_interest_expense_deduction = min_(
+            investment_interest_expense, net_investment_income
+        )  # Line 8
+        form_4952_amount = tax_unit(
+            "investment_income_form_4952", period
+        )  # Line 9
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:28]: make the line comment above of the line of code, e.g.:
```suggestion
        # Line 9
        form_4952_amount = tax_unit(
            "investment_income_form_4952", period
        )  
```

File: policyengine_us/variables/household/expense/person/investment_expense.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class investment_expense(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/expense/person/investment_expense.py:4]:
```suggestion
class investment_expenses(Variable):
```

File: policyengine_us/variables/household/expense/person/investment_expense.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    value_type = float
+    entity = Person
+    label = "Investment expense"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/expense/person/investment_expense.py:7]:
```suggestion
    label = "Investment expenses"
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/household/expense/person/investment_expense.py
Comment (MaxGhenis) [inline policyengine_us/variables/household/expense/person/investment_expense.py:1]: are we sure this isn't the same as the existing `interest_expense`? please check its uses against this one

Comment (vrathi101) [reply policyengine_us/variables/household/expense/person/investment_expense.py:1]: According to the CA form, "Investment expenses are your allowed deductions, other than interest expense, directly connected with the production of investment income." So I believe they're different, but I may be wrong.

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    def formula(tax_unit, period, parameters):
+        # Lines 2, 3, 4a, 4b, 4c, 4d, 4e, 4f, 7 from tax form not included in calculations
+        # Line 1
+        investment_interest_expense = add(
+            tax_unit, period, ["investment_interest_expense"]
+        )
+        # Line 5
+        investment_expenses = add(tax_unit, period, ["investment_expenses"])
+        # Line 6
+        net_investment_income = (
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:22]: income will not be the difference between two types of expenses

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        # )
+        net_investment_income = tax_unit("net_investment_income", period)
+        # Line 8
+        investment_interest_expense_deduction = min_(
+            investment_interest_expense, net_investment_income
+        )
+        # Line 9
+        form_4952_amount = tax_unit("investment_income_form_4952", period)
+        # Line 10
+        return np.absolute(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:33]:
```suggestion
        return np.absolute(
```

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        # Line 1
+        investment_interest_expense = add(
+            tax_unit, period, ["investment_interest_expense"]
+        )
+        # Line 5
+        # investment_expenses = add(tax_unit, period, ["investment_expenses"])
+        # Line 6
+        # net_investment_income = (
+        #     investment_interest_expense - investment_expenses
+        # )
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:24]: delete commented files

File: policyengine_us/variables/household/expense/person/investment_expense.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
+
+
+class investment_expenses(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/household/expense/person/investment_expense.py:4]: delete this file,

Net investment income seems to serve the purpose

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/excluded_deductions.yaml [hunk 8]
```diff
@@ -3,6 +3,7 @@ values:
   2021-01-01:
     - medical_expense_deduction
     - casualty_loss_deduction
+    - ca_investment_interest_deduction
 
 metadata:
   unit: variable
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/excluded_deductions.yaml:9]:
```suggestion
  unit: list
  period: year
```

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 9]
```diff
@@ -9,3 +9,26 @@ class ca_investment_interest_deduction(Variable):
     documentation = "https://www.ftb.ca.gov/forms/2021/2021-3526.pdf"
     definition_period = YEAR
     defined_for = StateCode.CA
+
+    def formula(tax_unit, period, parameters):
+        # Lines 2, 3, 4a, 4b, 4c, 4d, 4e, 4f, 7 from tax form not included in calculations
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:14]:
```suggestion
```

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 10]
```diff
@@ -9,3 +9,26 @@ class ca_investment_interest_deduction(Variable):
...
+            investment_interest_expense - investment_expenses
+        )
+        # Line 8
+        investment_interest_expense_deduction = min_(
+            investment_interest_expense, net_investment_income
+        )
+        # Line 9
+        form_4952_amount = tax_unit("investment_income_form_4952", period)
+        # Line 10
+        return np.absolute(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:32]:
```suggestion
        return np.abs(
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/excluded_deductions.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/ca/tax/income/deductions/itemized/limit/excluded_deductions.yaml:1]: should it be part of this list or a list of all actual deductions?

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py [hunk 11]
```diff
@@ -9,3 +9,23 @@ class ca_investment_interest_deduction(Variable):
     documentation = "https://www.ftb.ca.gov/forms/2021/2021-3526.pdf"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ca/tax/income/deductions/ca_investment_interest_deduction.py:9]: could you add a legal code reference?

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**